python/ccf/ledger.py

```diff
@@ -297,19 +520,81 @@ def __init__(self, directory: str):
             ),
         )
 
+        # Verifies there are no missing ledger chunks
+        self._verify_commited_files(sorted_ledgers)
+
         for chunk in sorted_ledgers:
             if os.path.isfile(os.path.join(directory, chunk)):
                 if not is_ledger_chunk_committed(chunk):
                     LOG.warning(f"The file {chunk} has not been committed")
                 self._filenames.append(os.path.join(directory, chunk))
 
+        # Initialize LedgerValidator isntance which will be passed to LedgerChunks.
+        self._ledger_validator = LedgerValidator()
+
     def __next__(self) -> LedgerChunk:
         self._fileindex += 1
         if len(self._filenames) > self._fileindex:
-            self._current_chunk = LedgerChunk(self._filenames[self._fileindex])
+            self._current_chunk = LedgerChunk(
+                self._filenames[self._fileindex], self._ledger_validator
+            )
             return self._current_chunk
         else:
             raise StopIteration
 
     def __iter__(self):
         return self
+
+    def _verify_commited_files(self, sorted_ledgers: list):
+        """
+        Verify that there are no missing ledger chunks.
+        Throws if there are missing files.
+        """
+
+        # First commit should always be 1
+        expected_commit = 1
+
+        # initialize end commit id to 1
+        end_commit_id = -1
+
+        for filename in sorted_ledgers:
+            match = re.search(self.LEDGER_COMMITTED_FILE_NAME_REGEX, filename)
+            if match:
+                start_commit_id = match.group(1)
+                if (len(match.group(2))) > 0:
```

Comment:
Python supports named groups (https://docs.python.org/3/library/re.html#regular-expression-syntax, with `(?P<name>...)` and `m.group('name')`. It's mildly more readable, and less prone to breaks when the regexp changes.